local/include/core/ProxyVerifier.h

```diff
@@ -202,148 +212,252 @@ class RuleCheck {
 
   /** Generate @a PresenceCheck, invoked by the factory function when the
    * "absence" flag is present.
    *
-   * @param name TextView holding the name of the target field
-   * @param value TextView (unused) in order to have the same signature as
+   * @param name The name of the target field
+   * @param value (unused) Used in order to have the same signature as
    * make_equality
    * @return A pointer to the Presence instance generated, holding a name
    * TextView for the rule to compare inputs to
    */
-  static std::shared_ptr<RuleCheck> make_presence(swoc::TextView name,
-                                                  swoc::TextView value);
+  static std::shared_ptr<RuleCheck> make_presence(
+      swoc::TextView name,
+      swoc::TextView value);
+
+  /**
+   * @param values (unusesd) The list of values specified in the YAML node.
+   */
+  static std::shared_ptr<RuleCheck> make_presence(
+      swoc::TextView name,
+      std::list<swoc::TextView> values);
 
   /** Generate @a AbsenceCheck, invoked by the factory function when the
    * "absence" flag is present.
    *
-   * @param name TextView holding the name of the target field
-   * @param value TextView (unused) in order to have the same signature as
+   * @param name The name of the target field
+   * @param value (unused) Used in order to have the same signature as
    * make_equality
    * @return A pointer to the AbsenceCheck instance generated, holding a name
    * TextView for the rule to compare inputs to
    */
-  static std::shared_ptr<RuleCheck> make_absence(swoc::TextView name,
-                                                 swoc::TextView value);
+  static std::shared_ptr<RuleCheck> make_absence(
+      swoc::TextView name,
+      swoc::TextView value);
+
+  /**
+   * @param values (unusesd) The list of values specified in the YAML node.
+   */
+  static std::shared_ptr<RuleCheck> make_absence(
+      swoc::TextView name,
+      std::list<swoc::TextView> values);
 
   /** Pure virtual function to test whether the input name and value fulfill the
    * rules for the test
    *
-   * @param key TextView The identifying transaction key.
-   * @param name TextView holding the name of the target field (null if not
-   * found)
-   * @param value TextView holding the value of the target field (null if not
-   * found)
-   * @return Whether the check was successful or not
+   * @param transaction_key The key identifying the transaction.
+   * @param name The name of the target field (null if not found).
+   * @param value The value of the target field (null if not found).
+   * @return Whether the check was successful.
    */
-  virtual bool test(swoc::TextView key, swoc::TextView name,
+  virtual bool test(swoc::TextView transaction_key, swoc::TextView name,
                     swoc::TextView value) const = 0;
+
+  virtual bool test(swoc::TextView transaction_key, swoc::TextView name,
+                    std::list<swoc::TextView> values) const = 0;
```

Comment:
Gah.

Reply:
Looking back on this, I think it's an error to pull the `TextView` out before passing it in. Why not just pass the YAML node and let the method detect scalar vs. list? That's what I tend to do.

Reply:
I see what you're saying, but that won't work here actually. These parameters to test() come in off the wire, not from the YAML file.